### DISCUSSION TITLE
How to enable/disable mopping? Here: X20+

### DISCUSSION THREAD

**Discussion by max-t-d** [al-one/hass-xiaomi-miot#2094] [Q&A]
Hi there,
first up: thanks for the work - I got the vacuum into HA absolutely no problem and can control it like a charm locally (works fine when taking away its internet connection so far).
However, after digging around a bit, I'm stuck on how to tell the vacuum whether to mop or not. I.e. where can I replicate the first row of options from xiaomi home app (vacuum / vacuum + mop / mop)? Second (vac power - although this is documented as read only???) and third row (mop mode) are very clear. Bit of a shame xiaomi doesn't allow to just set mop mode = 0 like roborock ;-)
Thus: how do I actually disable and enable the mopping via the action call? Do I go about this via piid: 10 / [clean-extend-data](https://miot-spec.org/miot-spec-v2/spec/property?type=urn:xiaomi-spec:property:clean-extend-data:0000000a) and if so, does anyone have any clue about what to actually put inside there?
Thanks a bunch in advance!

**Comment by al-one** [comment]
[xiaomi.vacuum.c102gl.js](https://github.com/user-attachments/files/18168581/xiaomi.vacuum.c102gl.js.txt)

Perhaps you can find some clues here.

**Reply by max-t-d** [reply]
Thanks, will have a look around!

**Comment by JulioVillamizar** [comment]
Hi, were you able to resolve this? I have the exact same problem and can't find a way to solve it.

**Reply by max-t-d** [reply]
Sadly I rather quickly gave up and switched to dreame vacuum integration by tasshack. Control via that isn’t local, which is annoying, but the control features are very complete…

**Comment by JulioVillamizar** [comment]
Okay, I installed the Dreame Vacuum integration by Tasshack, but I don't see those options. Do they appear for you? Or did you use a specific service?

I have an X20+

**Comment by max-t-d** [comment]
It’s a separate select entity called select.x20plus_cleaning_mode for me (x20+ as well), not part of the vacuum entity.

**Comment by JulioVillamizar** [comment]
No, my integration didn't create that entity. It's strange, maybe it's a firmware issue. My integration added the X20+ as xiaomi.vacuum.c102gl. I'm not sure if that's correct.

Device Information
xiaomi.vacuum.c102gl
by Dreametech™
Firmware: 4.5.6_1087

Was yours also added as xiaomi.vacuum.c102gl?

**Comment by max-t-d** [comment]
Yes, that is the correct model for mine, too, and the correct model number it is listed as in the miot docs as well. I’m I’m older firmware, but that really shouldn’t matter.
One thing I can think of: are your mop pads installed in the robot? The selection gets automatically limited to just vacuum if not, so maybe if they’re not installed the select doesn’t even get created in the first place.

**Comment by JulioVillamizar** [comment]
Hi, I finally figured out what the problem was. When I was setting up the Dreame integration, I had to configure device.py with this line.

result = self._protocol.get_properties(props[:5])

So the integration only configured 5 options for me. Now I modified it with

result = self._protocol.get_properties(props[:10])

And it works perfectly.

Thanks for your answers; without them, I wouldn't have been able to detect the problem.
